packages/plugin-ext/src/plugin/timeline.ts

```diff
@@ -0,0 +1,128 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { Plugin, TimelineExt, TimelineMain } from '../common';
+import { RPCProtocol } from '../common/rpc-protocol';
+import { Disposable, ThemeIcon } from './types-impl';
+import { PLUGIN_RPC_CONTEXT } from '../common';
+import { CancellationToken } from '@theia/core/lib/common';
+import { DisposableCollection } from '@theia/core/lib/common/disposable';
+import { URI } from 'vscode-uri';
+import { PluginIconPath } from './plugin-icon-path';
+import { CommandRegistryImpl } from './command-registry';
+import * as theia from '@theia/plugin';
+import { Timeline, TimelineItem, TimelineOptions } from '@theia/timeline/lib/common/timeline-protocol';
+
+export class TimelineExtImpl implements TimelineExt {
```

Comment:
Please make sure that code in the extension host process is a copy from code from VS Code.

All such files should have only adjusted imports, MS copyright and a link to code in VS Code repository which should be used next time to diff and pick up changes.

Obviously a CQ is required as well.

Reply:
done, created CQs: [CQ22324](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22324) [CQ22326](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22326) [CQ22327](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22327)

Reply:
One CQ will be enough?

Reply:
Created a new CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22336

Reply:
for some reasons cannot download attached zip file, maybe eclipse issue

Reply:
was able to download, it does not seem to preserve the structure, please see from referenced docs:
> We then need to attach the copied code (from original project) to the CQ, as a zip or tar archive. It should contain only the copied file(s) as well as file(s) containing potentially relevant license info. e.g. if we copied file foo.ts from directory src/node and there is not relevant license-related file in src, we could create the following archive to attach to the PR:
```
license.txt
src/node/foo.ts
```

Reply:
Reopened a new CQ with correct attachment: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22340